code/modules/lighting/lighting_turf.dm

```diff
@@ -88,6 +88,11 @@
 			else //If fulltile and opaque, then the whole tile blocks view, no need to continue checking.
 				directional_opacity = ALL_CARDINALS
 				break
+	else
+		for(var/atom/elem in src.contents)
+			for(var/datum/light_source/source in elem.light_sources)
+				if(!source?.effect_str)
+					source.vis_update()
```

Comment:
This cannot be correct. vis_update() should only be called by SSlighting.

Reply:
I can change this to `update_corners` like it was the first time, or to `force_update`